src/web/wms/src/main/java/org/geoserver/wms/web/data/StylePage.java

```diff
@@ -79,12 +79,7 @@ protected Component headerPanel() {
         header.add(removal = new SelectionRemovalLink("removeSelected", table, dialog) {
             @Override
             protected StringResourceModel canRemove(CatalogInfo object) {
-                StyleInfo s = (StyleInfo) object;
-                if ( StyleInfo.DEFAULT_POINT.equals( s.getName() ) || 
-                    StyleInfo.DEFAULT_LINE.equals( s.getName() ) || 
-                    StyleInfo.DEFAULT_POLYGON.equals( s.getName() ) || 
-                    StyleInfo.DEFAULT_RASTER.equals(s.getName()) ||
-                    StyleInfo.DEFAULT_GENERIC.equals(s.getName())) {
+                if (isDefaultStyle(object)) {
```

Comment:
Moving it to the catalog probably makes sense, would that fix it for the REST API as well? I didn't want to worry too much about the rest API, since I wanted to at least get this improvement out of the way and there are already plenty of differences between what the REST API does and the UI does.